### PR TITLE
Fix undo (CTRL + Z) and redo for inserting an entry

### PR DESCRIPTION
**Pull request by brunaoo**
Fixes https://github.com/JabRef/jabref/issues/9737

Removed default switch from EditAction class. This was causing an exception for the user when the Ctrl + Z command was issued.

```[tasklist]
### Compulsory checks
- [X] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [X] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [X] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [X] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.
```

### CODE REVIEW

**Comment by koppor** [comment]
Thank you for working on it. Please check your checkstyle setup at https://devdocs.jabref.org/getting-into-the-code/guidelines-for-setting-up-a-local-workspace. Scroll down...

**Comment by koppor** [comment]
For a quick fix, you can click on "Files" above and see Review Dog complaining about two spaces (which should be one). This should be an easy fix ^^.

**Comment by calixtus** [comment]
If the bug was introduced in 5.10 it is not necessary to document the fix in the changelog, since we only mention fixes from release to release:
grafik

However, this fix may remove the exception, but the problem is not fixed.
You killed the messenger, but did not fix the issue itself.

The real question is imho: Why is the Ctrl-Z keyboard event not consumed, but reaches the switch statement?

**Comment by koppor** [comment]
There should be a GUI test - if possible. There are other GUI tests at https://github.com/JabRef/jabref/tree/main/src/test/java/org/jabref/gui.

**Comment by Siedlerchr** [comment]
We took another look at this PR and found at least one root cause for the issue and at least input controls work now

File: src/main/java/org/jabref/gui/edit/EditAction.java [hunk 1]
```diff
@@ -75,7 +74,10 @@ public void execute() {
                     case CUT -> frame.getCurrentLibraryTab().cut();
                     case PASTE -> frame.getCurrentLibraryTab().paste();
                     case DELETE_ENTRY -> frame.getCurrentLibraryTab().delete(false);
-                    default -> throw new IllegalStateException("Only cut/copy/paste supported but got " + action);
+                    case REDO -> frame.getUndoManager().redo();
+                    case UNDO -> frame.getUndoManager().undo();
+                    default -> LOGGER.debug("Only cut/copy/paste supported but got: {} and focus owner {}", action, focusOwner);
+                    // default -> throw new IllegalStateException("Only cut/copy/paste supported but got " + action);
```
Comment (koppor) [inline src/main/java/org/jabref/gui/edit/EditAction.java:80]: Comment needs to be removed.

**Comment by calixtus** [comment]
Is the CannotUndoException catched?

**Comment by Siedlerchr** [comment]
Fixed the occuring of exception by checking for canRedo/canUndo

File: src/main/java/org/jabref/gui/undo/CountingUndoManager.java [hunk 2]
```diff
@@ -27,16 +27,20 @@ public synchronized boolean addEdit(UndoableEdit edit) {
...
     @Override
     public synchronized void undo() throws CannotUndoException {
-        super.undo();
-        current--;
-        postUndoRedoEvent();
+        if (canUndo()) {
+            super.undo();
+            current--;
+            postUndoRedoEvent();
+        }
```
Comment (koppor) [inline src/main/java/org/jabref/gui/undo/CountingUndoManager.java:34]: I think, the exception should be caught by the caller. On the one hand, JabRef should not enable Ctrl+Z if it is not possible to do undo something. On the other hand, we will surely eventually change all the undo/redo things. Anyone already said Redux/MobX / Command Pattern everywhere?

**Review by koppor — APPROVED** [review]

**Pull request merged**